### ISSUE TITLE
Jackett API documentation

### ISSUE THREAD

**Issue by Piccirello** [Jackett/Jackett#2706]
**Jackett version**:  All
**Mono version** (if not using Windows):  N/A

-------------------------------
Is there existing documentation for Jackett API v2? I can't seem to find any in the Wiki or via google-fu.

**Comment by userdocs** [comment]
Not that i could find, which is why i asked this https://github.com/Jackett/Jackett/issues/2690

I will try to understand it better and add some info here https://github.com/Jackett/Jackett/wiki

**Comment by kaso17** [comment]
Only the torznap/poto search API is documented.
The other APIs for configuration, etc. should be self explanatory from looking at the corresponding requests when using the web GUI.

What are you trying to to?

**Comment by edalex86** [comment]
What about some words in wiki about /server/config and /indexers endpoints?
1) how to switch json/xml output and what are proper headers for them? /Indexers is not working in browser since it's trying to convert json to xml somehow. Jackett receiving json but headers are almost the same as in browser/ I assume it could be done with Accept: application/json but Jackett doesn't have it.
2) Do we need to provide timestamp as parameter?
3) Is any other /endpoint in api?

**Comment by kaso17** [comment]
currently torznab only supports XML and all other APIs only support JSON. It's not possible to switch (pull requests to change this are welcome).
"Accept: application/json" or no Accept at all should fix the json APIs (browsers include xml by default in the accept, but no json).

timestamp is not needed (only added to avoid browser caching).
Other than /indexers? Yes. What do you need?

**Comment by edalex86** [comment]
Well, I'm still in development process so I don't know for sure. Continuing from other discussion maube I need categories tree to make fallback. Maybe tvsearch/movieseach support indication for indexers in json api or aggregate list of indexers with such info in torznab api.

**Comment by kaso17** [comment]
I would highly recommend using the toznab API.
If it helps I can have a look at putting together a list of all indexers including their caps (same as the q=caps output). Would that help?

**Comment by edalex86** [comment]
Yeah. It would be nice. I don't know for others but I'll need only configured ones.

**Comment by kaso17** [comment]
please try v0.8.862 (pre release)

**Comment by edalex86** [comment]
It's working, thanks! It has all indexers even not configured, is it right? So loading is not so fast.
Interesting fact that I can't see any site supporting tvdbid/tvmazeid in tv-search params like in api descrption from main page https://github.com/Sonarr/Sonarr/wiki/Implementing-a-Torznab-indexer
And probably my first auto-update didn't go well (jacketttray is crashing ;-D) and I'll reinstall Jackett later. Maybe I clicked on update button too many times. But that's because nothing happened visually :-D Don't accept this as bug report, please

**Comment by kaso17** [comment]
added a &configured=true/false argument to get either all configured or all non configured indexers only via 280547ad01c1d700b6b07527dfff690c3bcc718a

**Comment by kaso17** [comment]
tvmaze is dead.
Jackett currently doesn't handle tvdbid because the only indexer which could support it is rarbg. I'm not aware of any other torrent tracker supporting it.

**Comment by edalex86** [comment]
Omg, you're fast. Thanks!
BTN and HDBits supports tvdb for sure and as for tvmaze I think you're talking about tvrage which is "rid" I believe and tvmaze is living and growing.

**Comment by and7ey** [comment]
I am also missing an API documentation. How to search for movie of particular quality and in particular language? Can I search by imdb id? kinopoisk id? How to sort by seeders quantity? etc.

**Comment by ilike2burnthing** [comment]
See the readme starting from here - https://github.com/Jackett/Jackett#aggregate-indexers

**Comment by and7ey** [comment]
> See the readme starting from here - https://github.com/Jackett/Jackett#aggregate-indexers

thanks, but it doesn't answer my questions

**Comment by ilike2burnthing** [comment]
Quality and/or language should be included in the search term (assuming there are results with that information in the title). IMDB ID is detailed in the link above. Kinopoisk ID is not supported (full list of supported search IDs are detailed in the link above). Sorting, if available, is done in the indexer's settings. Other functions of the API are detailed in the link above.
